tests/test_flows.py

```diff
@@ -401,18 +401,86 @@ def test_channel_selection(self):
+
+        # payload_wlan2  - request for change channel on 36
+        payload_wlan2="0x14 {0x51 {0x00 0x00}} {0x52 {0x00 0x00}} {0x53 {0x00 0x00}} {0x54 {0x00 0x00}} {0x73 0x03 {0x28 0x2C 0x30} 0x00} {0x74 0x01 {0x2C} 0x00} {0x75 {0x00 0x00}} {0x76 {0x00 0x00}} {0x77 {0x00 0x00}} {0x78 {0x00 0x00}} {0x79 {0x00 0x00}} {0x7A {0x00 0x00}} {0x7B {0x00 0x00}} {0x7C {0x00 0x00}} {0x7D {0x00 0x00}} {0x7E {0x00 0x00}} {0x7F {0x00 0x00}} {0x80 0x05 {0x3A 0x6A 0x7A 0x8A 0x9B} 0x00} {0x81 {0x00 0x00}} {0x82 {0x00 0x00}}"
+
+        """
+        Step 1: Trigger channel selection to channel 6 and 36. Check that
+                operating channel report was sent.
+
+        Step 2: Trigger channel selection to channel 6 and 36 again - check that
+                operating channel report is sent again. This is to catch bugs
+                when we don't send channel report when there is no need to
+                switch channel
+        """
+        for i in range(1, 3):
+            transmit_power = int(tp20dBm,16)
+            self.debug("Send channel selection request, step %i" % i)
+            self.gateway_ucc.dev_send_1905(
+                self.mac_repeater1,
+                0x8006,
+                tlv(0x8B,0x005F, f'{self.mac_repeater1_wlan0} {payload_wlan0}'),
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan0} {tp20dBm}'),
+                tlv(0x8B,0x004C, f'{self.mac_repeater1_wlan2} {payload_wlan2}'),
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan2} {tp20dBm}')
+            )
+            time.sleep(1)
+
+            self.debug("Confirming channel selection request has been received on agent, step %i" % i)
+
+            self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+
+            self.debug("Confirming tlvTransmitPowerLimit has been received with correct value on agent, step %i" % i)
+            self.check_log(self.repeater1, "agent_wlan0", "tlvTransmitPowerLimit %d" % transmit_power)
+            self.check_log(self.repeater1, "agent_wlan0", "ACTION_APMANAGER_HOSTAP_CHANNEL_SWITCH_ACS_START")
+            self.check_log(self.repeater1, "agent_wlan2", "tlvTransmitPowerLimit %d" % transmit_power)
+            self.check_log(self.repeater1, "agent_wlan2", "ACTION_APMANAGER_HOSTAP_CHANNEL_SWITCH_ACS_START")
+
+            self.debug("Confirming ACK_MESSAGE has been received on agent, step %i" % i)
+            self.check_log(self.repeater1, "agent_wlan0", "ACK_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan2", "ACK_MESSAGE")
```

Comment:
Which ACK_MESSAGE?

For an ACK, we should definitely always check the MID because there are so many acks that it will always hit one.